src/server/chat/systems.rs

```diff
@@ -0,0 +1,49 @@
+use std::time::UNIX_EPOCH;
+
+use crate::prelude::*;
+
+pub fn handle_network_chat_message_send(
+    mut server: ResMut<RenetServer>,
+    mut player_send_messages: EventReader<chat_events::PlayerChatMessageSendEvent>,
+    mut chat_messages: ResMut<chat_resources::ChatHistory>,
+) {
+    for event in player_send_messages.read() {
+        let message = event.message.clone();
+        let client_id = event.client_id;
+        let message_count = chat_messages.messages.len();
+        let message_id = message_count;
+
+        chat_messages.messages.push(lib::ChatMessage {
+            client_id,
+            message_id,
+            message,
+            timestamp: get_current_time_in_ms(),
+        });
+
+        let response_message =
+            lib::NetworkingMessage::ChatMessageSync(chat_messages.messages.clone());
```

Comment:
Do you really have to send back the whole chat history to the player?

Reply:
Also no maximum size for this array?

Reply:
It's a dynamic vector, the chat message history can get infinitely long (until the transport layer can't take it anymore). I could restrict it to the last 255 messages (i think its a nice number).

Reply:
No, only need to sync the chat history for new players.

Reply:
Currently working on refactoring that.